lib/wcc/contentful/content_type_indexer.rb

```diff
@@ -0,0 +1,153 @@
+            content_type.is_a?(Contentful::Management::ContentType)
+          create_type(content_type.id, content_type.fields)
+        else
+          create_type(content_type.dig('sys', 'id'), content_type['fields'])
+        end
+
+      @types[content_type[:name]] = content_type
+    end
+
+    def create_type(content_type_id, fields)
+      content_type = {
+        name: constant_from_content_type(content_type_id),
+        content_type: content_type_id,
+        fields: {}
+      }
+
+      fields.each do |f|
+        field = create_field(f)
+        content_type[:fields][field[:name]] = field
+      end
+
+      content_type
+    end
+
+    # hardcoded because the Asset type is a "magic type" in their system
+    def create_asset_type
+      {
+        name: 'Asset',
+        content_type: 'Asset',
+        fields: {
+          'title' => { name: 'title', type: :String },
+          'description' => { name: 'description', type: :String },
+          'file' => { name: 'file', type: :Json }
+        }
+      }
+    end
+
+    private
+
+    def create_field(field)
+      if field.respond_to?(:raw)
+        create_field_from_raw(field.raw)
+      elsif field.respond_to?(:to_h)
+        create_field_from_raw(field.to_h)
+      else
+        create_field_from_managed(field)
+      end
+    end
+
+    def create_field_from_managed(managed_field)
+      field = {
+        name: managed_field.id,
+        type: find_field_type(managed_field),
+        required: managed_field.required
+      }
+      field[:array] = true if managed_field.type == 'Array'
+
+      if field[:type] == :Link
+        validations =
+          if field[:array]
+            managed_field.items.validations
+          else
+            managed_field.validations
+          end
+        field[:link_types] = resolve_managed_link_types(validations) if validations.present?
+      end
+      field
+    end
+
+    def create_field_from_raw(raw_field)
+      field_name = raw_field['id']
+      field = {
+        name: field_name,
+        type: find_field_type(raw_field),
+        required: raw_field['required']
+      }
+      field[:array] = true if raw_field['type'] == 'Array'
+
+      if field[:type] == :Link
+        validations =
+          if field[:array]
+            raw_field.dig('items', 'validations')
+          else
+            raw_field['validations']
+          end
+        field[:link_types] = resolve_raw_link_types(validations) if validations.present?
+      end
+      field
+    end
+
+    def find_field_type(field)
+      # 'Symbol' | 'Text' | 'Integer' | 'Number' | 'Date' | 'Boolean' |
+      #  'Object' | 'Location' | 'Array' | 'Link'
+      case raw_type = field.try(:type) || field['type']
+      when 'Symbol', 'Text'
+        :String
+      when 'Integer'
+        :Int
+      when 'Number'
+        :Float
+      when 'Date'
+        :DateTime
+      when 'Boolean'
+        :Boolean
+      when 'Object'
+        :Json
+      when 'Location'
```

Comment:
still `Location`?

Reply:
I thought you hadn't updated your intermediate fixture yet...

Reply:
'Location' is Contentful's name for this data type.  Here is where we translate that to 'coordinates'.